core/commands/pin.go

```diff
@@ -114,42 +115,33 @@ var addPinCmd = &cmds.Command{
 				case <-ticker.C:
 					out <- &AddPinOutput{Progress: v.Value()}
 				case <-ctx.Done():
-					res.SetError(ctx.Err(), cmds.ErrNormal)
+					res.SetError(ctx.Err(), cmdsutil.ErrNormal)
 					return
 				}
 			}
 		}()
 	},
 	Marshalers: cmds.MarshalerMap{
 		cmds.Text: func(res cmds.Response) (io.Reader, error) {
+			v := unwrapOutput(res.Output())
+
 			var added []string
 
-			switch out := res.Output().(type) {
+			switch out := v.(type) {
 			case *AddPinOutput:
-				added = out.Pins
-			case <-chan interface{}:
-				progressLine := false
-				for r0 := range out {
-					r := r0.(*AddPinOutput)
-					if r.Pins != nil {
-						added = r.Pins
-					} else {
-						if progressLine {
-							fmt.Fprintf(res.Stderr(), "\r")
-						}
-						fmt.Fprintf(res.Stderr(), "Fetched/Processed %d nodes", r.Progress)
-						progressLine = true
-					}
-				}
-				if progressLine {
-					fmt.Fprintf(res.Stderr(), "\n")
+				if out.Pins != nil {
+					added = out.Pins
+				} else {
+					fmt.Fprintf(res.Stderr(), "Fetched/Processed %d nodes\r", out.Progress)
```

Comment:
That is implicit because `out.Pins` is non-nil if the progress option is disabled